persistence/src/main/scala/com/lightbend/lagom/internal/persistence/cassandra/CassandraReadSideHandler.scala

```diff
@@ -0,0 +1,172 @@
+  cassandraProcessor: CassandraReadSideProcessor[Event],
+  dispatcher:         String
+)(implicit ec: ExecutionContext) extends CassandraReadSideHandler[Event, BiFunction[_ <: Event, UUID, CompletionStage[JList[BoundStatement]]]](
+  session,
+  cassandraProcessor.defineEventHandlers(new cassandraProcessor.EventHandlersBuilder).handlers,
+  dispatcher
+) {
+
+  override protected def invoke(handler: BiFunction[_ <: Event, UUID, CompletionStage[JList[BoundStatement]]], event: Event, offset: Offset): CompletionStage[JList[BoundStatement]] = {
+    offset match {
+      case uuid: TimeBasedUUID => handler.asInstanceOf[BiFunction[Event, UUID, CompletionStage[JList[BoundStatement]]]].apply(event, uuid.value)
+      case other               => throw new IllegalArgumentException("CassandraReadSideProcessor does not support offsets of type " + other.getClass.getName)
+    }
+  }
+
+  override def prepare(tag: AggregateEventTag[Event]): CompletionStage[Offset] = {
+    Future.successful(cassandraProcessor).flatMap(_.prepare(session).toScala).map { maybeUuid =>
+      if (maybeUuid.isPresent) {
+        Offset.timeBasedUUID(maybeUuid.get)
+      } else Offset.NONE
+    }.toJava
+  }
+}
+
+private[cassandra] class OffsetStore(session: CassandraSession, offsetTable: String)(implicit ec: ExecutionContext) {
+  private var writeOffsetStatement: PreparedStatement = _
+
+  def globalPrepare = {
+    session.executeCreateTable(s"""
+                                  |CREATE TABLE IF NOT EXISTS $offsetTable (
+                                  |  partition text, timeUuidOffset timeuuid, sequenceOffset bigint,
+                                  |  PRIMARY KEY (partition)
+                                  |)""".stripMargin).toScala
+  }
+
+  def prepare(partition: String) = {
+    prepareWriteOffset
+      .flatMap(_ => readOffset(partition))
+  }
+
+  private def prepareWriteOffset: Future[Done] = {
+    session.prepare(s"INSERT INTO $offsetTable (partition, timeUuidOffset, sequenceOffset) VALUES (?, ?, ?)").toScala.map { ps =>
+      writeOffsetStatement = ps
+      Done.getInstance()
+    }
+  }
+
+  private def readOffset(partition: String): Future[Offset] = {
+    session.selectOne(s"SELECT timeUuidOffset, sequenceOffset FROM $offsetTable WHERE partition = ?", partition).toScala.map(extractOffset)
+  }
+
+  private def extractOffset(maybeRow: Optional[Row]): Offset = {
+    if (maybeRow.isPresent) {
+      val row = maybeRow.get()
+      val uuid = row.getUUID("timeUuidOffset")
+      if (uuid != null) {
+        Offset.timeBasedUUID(uuid)
+      } else {
+        if (row.isNull("sequenceOffset")) {
+          Offset.NONE
+        } else {
+          Offset.sequence(row.getLong("sequenceOffset"))
+        }
+      }
+    } else Offset.NONE
+  }
+
+  def writeOffset(partition: String, offset: Offset): BoundStatement = {
+    offset match {
+      case Offset.NONE                => writeOffsetStatement.bind(partition, null, null)
```

Comment:
It can't be. `prepareWriteOffset` is always executed before event processing starts.  If `prepareWriteOffset` succeeds, it can't be null, if `prepareWriteOffset` fails, event processing will never start (event processing is done by an actor, all failures are fed back to the actor which crashes if it receives one, and is supervised by an exponential backoff supervisor that uses the stop strategy).